agentendpoint/agentendpoint.go

```diff
@@ -43,26 +49,59 @@ type Client struct {
 	raw *agentendpoint.Client
 }
 
-var client *Client
-
 // NewClient a new agentendpoint Client.
 func NewClient(ctx context.Context) (*Client, error) {
-	c, err := agentendpoint.NewClient(ctx)
+	opts := []option.ClientOption{
+		option.WithoutAuthentication(),
+		option.WithGRPCDialOption(grpc.WithTransportCredentials(credentials.NewTLS(nil))),
+		option.WithEndpoint(config.SvcEndpoint()),
+	}
+	logger.Debugf("Creating new agentendpoint client.")
+	c, err := agentendpoint.NewClient(ctx, opts...)
 	if err != nil {
 		return nil, err
 	}
-	client = &Client{raw: c}
-	return client, nil
+	return &Client{raw: c}, nil
+}
+
+// Close closes the underlying ClientConn.
+func (c *Client) Close() error {
+	return c.raw.Close()
 }
 
-// ReportTaskProgress calls the agentendpoint service ReportTaskProgress call injecting the InstanceIdToken.
-func (c *Client) ReportTaskProgress(ctx context.Context, req *agentendpointpb.ReportTaskProgressRequest) (res *agentendpointpb.ReportTaskProgressResponse, err error) {
-	logger.Debugf("ReportTaskProgress request:\n%s", util.PrettyFmt(req))
+func (c *Client) reportTaskStart(ctx context.Context) (res *agentendpointpb.ReportTaskStartResponse, err error) {
```

Comment:
s/reportTaskStart/ReportTaskStart

Reply:
I made these private as there's no need to access them outside the package